### PR TITLE
chore: unify & clean Cargo.toml versions and such

### PR DESCRIPTION
**Pull request by dpc**
Not everything, since I'm afraid to do it automatically, and I ran out patience.

### CODE REVIEW

File: crypto/aead/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "fedimint-aead"
-version = "0.4.0-alpha"
+version = { workspace = true }
```
Comment (dpc) [inline crypto/aead/Cargo.toml:3]: All of these are now `{ workspace = true }`, at least.

File: devimint/Cargo.toml [hunk 2]
```diff
@@ -52,4 +52,4 @@ fedimint-portalloc = { path = "../utils/portalloc" }
 fs-lock = "0.1.3"
 
 [build-dependencies]
-fedimint-build = { version = "=0.4.0-alpha", path = "../fedimint-build" }
+fedimint-build = { path = "../fedimint-build" }
```
Comment (elsirion) [inline devimint/Cargo.toml:55]: Won't that break publishing? I had to painstakingly add all these version numbers in addition to paths when first publishing to crates.io :sob:

**Review by elsirion — COMMENTED** [review]
Can we dry-run publishing? Alternatively: how about just publishing this rather random alpha version :sweat_smile: it won't be pulled in automatically by tooling imo since it's a pre-release, so not too bad imo.

File: devimint/Cargo.toml (hunk 2 above)
Comment (dpc) [reply devimint/Cargo.toml:55]: Oh. I thought I've seen plenty of ones that didn't have it ...

**Comment by dpc** [comment]
My bad. Will redo.

**Comment by dpc** [comment]
https://github.com/fedimint/fedimint/pull/5354

File: devimint/Cargo.toml (hunk 2 above)
Comment (elsirion) [reply devimint/Cargo.toml:55]: Maybe some tooling will add it automagically or I was just holding it wrong?

**Pull request closed without merging**